byteps/common/compressor/utils.h

```diff
@@ -214,6 +214,35 @@ unsigned int EliasDeltaDecode(BitReader<T>& bit_reader) {
   return num;
 }
 
+template <typename T, class F = std::function<bool(T)>>
+T HyperParamFinder(const kwargs_t& kwargs, std::string name,
+                   bool optional = false, F&& check = [](T) { return true; }) {
```

Comment:
```suggestion
template <typename T>
T HyperParamFinder(const kwargs_t& kwargs, const std::string& name,
                   bool optional = false, const std::function<bool(T)>& check = [](T) { return true; }) {
```

Reply:
What's the benefits of doing this?

Reply:
Well, this modification is made just because I think your former function definition is little bit complex. Sure, you can just ignore it cause this is not of high priority.

BTW, I can see that in your codes you made full use of `std::function` for lambda calls or static functions which will produce sort of overheads(see 45:27 at [CppCon2016, JasonTurner](https://www.youtube.com/watch?v=uzF4u9KgUWI)). I suggest that we can have those not-so-important coding enhancements when other high-priority tasks are done.

Reply:
Thanks for your suggestions. I will think about it.